### DISCUSSION TITLE
Cannot create btSoftBody rope smaller than length of one unit

### DISCUSSION THREAD

**Discussion by myint** [bulletphysics/bullet3#3486] [General]
Is there some minimum length for the soft body created by "btSoftBodyHelpers::CreateRope"? If so, how can I modify this? See below for the context.

It seems like the rope can't get any shorter than a length of 1 unit or so. For example, if I call "btSoftBodyHelpers::CreateRope()" with vectors a distance of 10 units apart, I can clearly see that the rope is longer than a rope created by with vectors a distance of 1 unit apart. But if I can't seem to make the rope any shorter than that. No matter how small the distance between the vectors are, the rope always remains the same length as if I used vectors a distance of 1 unit apart.

Below is an example using the `SoftBody` demo distributed with Bullet. I only modified the `Init_Ropes()` function.

With `btVector`s distance 5 units apart:

![5](https://cloud.githubusercontent.com/assets/1235108/6947156/2b4225e0-d85a-11e4-89b6-d8ab21c30e13.png)

With `btVector`s distance 1 unit apart:

![1](https://cloud.githubusercontent.com/assets/1235108/6947166/3bf35288-d85a-11e4-8dfb-dc9b06a86b74.png)

With `btVector`s distance 0.1 unit apart. Note that the rope is no shorter than when I had the vectors 1 unit apart (in the above example), which seems strange to me:

![0_1](https://cloud.githubusercontent.com/assets/1235108/6947168/40e36f26-d85a-11e4-9bbf-0ecc2d6de86f.png)

I previously mentioned this in the [forum](http://bulletphysics.org/Bullet/phpBB3/viewtopic.php?f=9&t=10474).

Thanks

**Comment by erwincoumans** [comment]
I haven't tested that feature, the btSoftBodyHelpers might have issues. If you end up fixing it, please share a patch.
